src/lib.rs

```diff
@@ -128,6 +135,24 @@ impl Cloudinary {
         Ok(json)
     }
 
+    /// uploads an image
+    /// ```rust
+    /// use cloudinary::{Source, Cloudinary};
+    /// use cloudinary::upload::{UploadOptions};
+    /// let options = UploadOptions::new().set_public_id("file.jpg".to_string());
+    /// let cloudinary = Cloudinary::new("api_key".to_string(), "cloud_name".to_string(), "api_secret".to_string() );
+    /// let result = cloudinary.upload_image(Source::Path("./image.jpg".into()), &options);
+    /// ```
+    pub async fn upload_image(
+        &self,
+        src: Source,
+        options: &UploadOptions<'_>,
+    ) -> Result<UploadResult> {
+        let new_opts = options.clone();
+        let updated_options = new_opts.set_resource_type(ResourceTypes::Image);
```

Comment:
If options already contain resource type, we will silently overwrite it, which is a potential bug in a client application. We can catch it on compile time.

Reply:
Yeah, I thought since the person is using the upload_image method, then the resource_type will have to be an image whether they set it or not

Reply:
We can let them know that the overwrite will happen in the docs

Reply:
How I image this API:
```rust
cloudinary.upload(Options::Image(...image_specific_options...))->UploadResultEnum;

cloudinary.upload_image(...image_specific_options...)->...image_specific_result
cloudinary.upload_audio(...audio_specific_options...)->...audio_specific_result
```

Reply:
Yeah, I get this. So I've used the official Python SDK, and it doesn't have an `upload_audio` method, it just has an `upload_image` and an `upload` method that can still be used to upload images.

Reply:
We'll probably need to determine the return type based on the values that are passed in as options. Hmmm.

Reply:
I'm currently participating in a Hackathon that's ending Wednesday. Please I'll tackle this after.